### ISSUE TITLE
RCS1214 not working for methods that return strings.

### ISSUE THREAD

**Issue by mahgo** [JosefPihrt/Roslynator#904]
**Product and Version Used**: 
Roslynator.Analyzers 4.1.0

**Steps to Reproduce**:
```csharp
string str = $"{MethodThatReturnsString()}";
```
**Actual Behavior**:
Rule not detected.

**Expected Behavior**:
Rule detected and suggested correction:
```csharp
string str = MethodThatReturnsString();
```

**Comment by JosefPihrt** [comment]
This transformation is not safe because interpolated string turns null into empty string, please see #446.

**Comment by mahgo** [comment]
@JosefPihrt Wouldn't the following example on RCS1214 also have this issue?:

### Code with Diagnostic

```csharp
$"{x}"
```

### Code with Fix

```csharp
x
```

**Comment by mahgo** [comment]
Do you think there might be value in having this as a separate analyzer then?

**Comment by JosefPihrt** [comment]
`x` in the example must be non-null string constant. I'll update docs accordingly.

Regarding your question: I don't think it would be valuable to add new analyzer because the code fix should produce semantically equivalent syntax which is not the case here.
